rust-semantics/syntax.md

```diff
@@ -0,0 +1,1040 @@
+would force us to support more Rust features in the main semantics.
+
+```k
+module RUST-SYNTAX
+    imports RUST-SHARED-SYNTAX
+
+    syntax Underscore ::= "_"  [token]
+
+```
+
+  https://doc.rust-lang.org/reference/identifiers.html
+
+```k
+
+  // TODO: Not implemented properly
+  syntax Identifier ::= r"[A-Za-z_][A-Za-z0-9\\_]*"  [prec(2), token]
+endmodule
+
+module RUST-SHARED-SYNTAX
+  imports STRING-SYNTAX
+```
+
+  https://doc.rust-lang.org/reference/crates-and-source-files.html
+
+```k
+    syntax Crate ::= InnerAttributes Items  [symbol(crate)]
+    syntax InnerAttributes ::= List{InnerAttribute, ""}  [symbol(innerAttributes)]
+    syntax Items ::= List{Item, ""}  [symbol(items)]
+
+```
+
+  https://doc.rust-lang.org/reference/attributes.html
+
+```k
+
+    syntax InnerAttribute ::= "#![" Attr "]"  [symbol(innerAttribute)]
+    syntax OuterAttribute ::= "#[" Attr "]"  [symbol(outerAttribute)]
+
+    syntax Attr ::= Identifier
+                  | Identifier "(" Identifier ")"
+                  | Identifier "(" StringLiteral ")"
+                  | Identifier "::" Identifier
+
+```
+
+  https://doc.rust-lang.org/reference/items.html
+
+```k
+
+  syntax Item ::= OuterAttributes VisOrMacroItem  [symbol(item)]
+  syntax NonEmptyOuterAttributes ::= OuterAttribute | OuterAttribute NonEmptyOuterAttributes
+  syntax OuterAttributes ::= "" | NonEmptyOuterAttributes
+  syntax VisOrMacroItem ::= VisItem | MacroItem
+  syntax MacroItem ::= MacroInvocationSemi | MacroRulesDefinition
+  syntax VisItem ::= MaybeVisibility VisItemItem
+  syntax MaybeVisibility ::= "" | Visibility
+  syntax VisItemItem  ::= Module
+                        | ExternCrate
+                        | UseDeclaration
+                        | Function
+                        | TypeAlias
+                        | Struct
+                        | Enumeration
+                        | Union
+                        | ConstantItem
+                        | StaticItem
+                        | Trait
+                        | Implementation
+                        | ExternBlock
+
+```
+
+  https://doc.rust-lang.org/reference/visibility-and-privacy.html
+
+```k
+
+  syntax Visibility  ::= "pub"
+                      | "pub" "(" "crate" ")"
+                      | "pub" "(" "self" ")"
+                      | "pub" "(" "super" ")"
+                      | "pub" "(" "in" SimplePath ")"
+
+```
+  https://doc.rust-lang.org/reference/paths.html#simple-paths
+```k
+
+  syntax SimplePath ::= SimplePathList | "::" SimplePathList
+  syntax SimplePathList ::= SimplePathSegment
+  syntax SimplePathList ::= SimplePathSegment "::" SimplePathList
+  syntax SimplePathSegment ::= Identifier | "super" | "self" | "crate" | "$crate"
+
+```
+  https://doc.rust-lang.org/reference/items/modules.html
+```k
+
+  syntax Module ::= MaybeUnsafe "mod" Identifier ";"
+                  | MaybeUnsafe "mod" Identifier "{" InnerAttributes Items "}"
```

Comment:
A comment rather than a review. 
`InnerAttributes Items` is `crate`, not sure why the definition in rust reference book is not using `crate` here.